code/modules/clothing/clothing.dm

```diff
@@ -542,6 +562,32 @@
 	icon_state = "[icon_state]_opentoe"
 	item_state = "[item_state]_opentoe"
 
+/obj/item/clothing/shoes/AltClick(mob/user)
+	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user))
+		return
+
+	if(!knife_slot)
+		return
+	if(!hidden_blade)
+		to_chat(user, "<span class='warning'>There's nothing in your [name]!</span>")
+		return
+
+	if(!isnull(user.get_active_hand()) && !isnull(user.get_inactive_hand()))
```

Comment:
```suggestion
	if(user.get_active_hand() && user.get_inactive_hand())
```
Just check if there's a value here, no need for a double not operator. this also really shouldn't be hand dependent, just force it on the floor if their hands are full

Reply:
I'm copying the same drawing logic from Holsters, which also require an open hand. Making a dexterous movement like drawing a weapon should require an open hand - I wont be budging on that one.